### ISSUE TITLE
Debug Branch: Assertion violation math/lp/lar_solver.cpp: 1706

### ISSUE THREAD

**Issue by wintered** [Z3Prover/z3#3483]
File: [bug.smt2](https://github.com/Z3Prover/z3/files/4367953/bug.txt) 
```
% z3debug-8d39694 bug.smt2
ASSERTION VIOLATION
File: ../src/math/lp/lar_solver.cpp
Line: 1706
all_vars_are_registered(coeffs)
(C)ontinue, (A)bort, (S)top, (T)hrow exception, Invoke (G)DB
ASSERTION VIOLATION
File: ../src/util/old_vector.h
Line: 370
idx < size()
(C)ontinue, (A)bort, (S)top, (T)hrow exception, Invoke (G)DB
```
OS: Ubuntu 18.04
Commit: 8d39694

**Comment by NikolajBjorner** [comment]
this one reproduces well

**Comment by levnach** [comment]
Fixed in 3e172733c594535eaba759cc4391e735dc755eea
